pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1484 @@
+                            )
+                        for i in range(0, len(vl['indices'])):
+                            ln = ln.replace(
+                                '__I_PLACEHOLDER_8675309_GROUP_%s_%s__'
+                                % (vl['indices'][i], ky),
+                                indexNames[i],
+                            )
+                    else:
+                        for i in range(0, len(vl['indices'])):
+                            ln = ln.replace(
+                                '__I_PLACEHOLDER_8675309_GROUP_%s_%s__'
+                                % (vl['indices'][i], ky),
+                                alphabetStringGenerator(indexCounter, True),
+                            )
+                            indexCounter += 1
+                else:
+                    for i in range(0, len(vl['indices'])):
+                        ln = ln.replace(
+                            '__I_PLACEHOLDER_8675309_GROUP_%s_%s__'
+                            % (vl['indices'][i], ky),
+                            alphabetStringGenerator(indexCounter, True),
+                        )
+                        indexCounter += 1
+
+        latexLines[jj] = ln
+
+    pstr = '\n'.join(latexLines)
+
+    vrIdx = 0
+    new_variableMap = ComponentMap()
+    for i in range(0, len(variableList)):
+        vr = variableList[i]
+        vrIdx += 1
+        if isinstance(vr, ScalarVar):
+            new_variableMap[vr] = vr.name
+        elif isinstance(vr, IndexedVar):
+            new_variableMap[vr] = vr.name
+            for sd in vr.index_set().data():
+                # vrIdx += 1
+                sdString = str(sd)
+                if sdString[0] == '(':
+                    sdString = sdString[1:]
+                if sdString[-1] == ')':
+                    sdString = sdString[0:-1]
+                new_variableMap[vr[sd]] = vr[sd].name
+        else:
+            raise DeveloperError(
+                'Variable is not a variable.  Should not happen.  Contact developers'
+            )
+
+    pmIdx = 0
+    new_parameterMap = ComponentMap()
+    for i in range(0, len(parameterList)):
+        pm = parameterList[i]
+        pmIdx += 1
+        if isinstance(pm, ScalarParam):
+            new_parameterMap[pm] = pm.name
+        elif isinstance(pm, IndexedParam):
+            new_parameterMap[pm] = pm.name
+            for sd in pm.index_set().data():
+                # pmIdx += 1
+                sdString = str(sd)
+                if sdString[0] == '(':
+                    sdString = sdString[1:]
+                if sdString[-1] == ')':
+                    sdString = sdString[0:-1]
+                new_parameterMap[pm[sd]] = str(pm[sd])  # .name
+        else:
+            raise DeveloperError(
+                'Parameter is not a parameter.  Should not happen.  Contact developers'
+            )
+
+    for ky, vl in new_variableMap.items():
+        if ky not in ComponentSet(latex_component_map.keys()):
```

Comment:
I don't think you need to make a ComponentSet here--can just do `if ky not in latex_component_map.keys()`

Reply:
Im leaving these because they work at present and I'm unsure about the suggested change.  But I believe I needed the component set to do equality checking at some point in the function